src/GifWin/MainWindow.xaml.cs

```diff
@@ -80,5 +81,15 @@ private void OnWindowKeyUp (object sender, KeyEventArgs e)
             if (e.Key == Key.Escape)
                 Hide();
         }
+
+        private void SearchBoxKeyPressed(object sender, KeyEventArgs e)
+        {
+            if (SearchStates.CurrentState == null || SearchStates.CurrentState.Name != "Searching")
```

Comment:
I'm actually not sure if this is 100% correct, but it seems that SearchStates.CurrentState is not set to the default state name, it's only set when you explicitly state transition, so checking for null is the equivalent here.